### ISSUE TITLE
Integrate has undefined name 'dim'

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#2833]
https://github.com/pydata/xarray/blob/master/xarray/core/dataset.py#L4085

Should that be called `coord` or `dim`? Currently there's a variable that's undefined:

```python
            raise ValueError('Coordinate {} does not exist.'.format(dim))
```
I would have made a quick fix but not sure the correct name

**Comment by shoyer** [comment]
I think it should be `coord`. Also we should add a unit test :)

**Comment by max-sixty** [comment]
Closed by #3010
